pulumi/terraform-provider-aws
# upstream v3.46.0

- inital commit
- sweeper
- docs
- changelog
- fmt
- docs fmt
- Update website/docs/r/budgets_budget_action.html.markdown
- Update website/docs/r/budgets_budget_action.html.markdown
- add waiters
- Update CHANGELOG.md after v3.45.0
- Add missing v3.45.0 CHANGELOG entry (#19555).
- r/aws_launch_configuration: Documentation category is 'Autoscaling' (as for the corresponding data source).
- Add EC2 resources missing from issue labeler action.
- Use shared naming logic for ASG and Launch Configuration (name_prefix fix)
- .github/workflows: Fix waiting-response label removal
- Add missing 'status' attribute to organizations datasource docs
- added virtual service data source
- added note for changelog
- fix terrafmt errors
- fixed terrafmt error
- 'name_prefix' is Computed.
- fix awsproviderlint error: 'XAT001: missing ErrorCheck'.
- Docs use incorrect argument for on_premises_instance_tag_filter
- add mount_options and validations
- add support for update
- changelog
- docs
- add support for config
- tests
- fmt
- disappears
- finder/waiter/fix flatten
- docs
- docs
- changelog
- fmt
- doc fmt
- Add a cautionary note for CMK-encrypted volumes (#12508)
- fix docs: SSO credentials is now supported (#19770)
- fix on_demand_percentage_above_base_capacity default value to 100 (#19685)
- Clarify key policy vs IAM policy (#19271)
- aws_iam_policy_document.statement.principals.type allows asterisk as value (#19170)
- chore: add codeowners to eventbridge (#19126)
- remove defaults, and make whitespace consistent (#19791)
- Apply suggestions from code review
- r/aws_budgets_budget_action: Tweak waiter statuses.
- Tweak AppMesh data source acceptance test names.
- Update CHANGELOG.md (Manual Trigger)
- build(deps): bump github.com/aws/aws-sdk-go in /awsproviderlint
- build(deps): bump github.com/aws/aws-sdk-go from 1.38.59 to 1.38.61
- add unit and refactor
- docs
- changelog
- extra test
- EKS node group scaling_config.min_size can be 0
- resource/aws_route53_resolver_firewall_config: Add new resource
- resource/aws_route53_resolver_firewall_config: Address review comments
- resource/aws_route53_resolver_firewall_config: Address review comments II
- r/aws_route53_resolver_firewall_config: Reference 'resourceAwsRoute53ResolverFirewallConfigUpdate'.
- fix idp update + more tests
- changelog
- fix changelog :)
- Check for nil values
- Adjust default timeouts to be aligned with documentation of the resource.
- Adjust default Glue timeouts to be aligned with documentation of the resource.
- add release note entry
- r/aws_route53_resolver_firewall_config: Tweak 'testAccCheckRoute53ResolverFirewallConfigDestroy'.
- r/aws_lambda_event_source_mapping: Additional IAM eventual consistency error message.
- Add CHANGELOG entry.
- Fix data source title and partial information (#19830)
- r/aws_sqs_queue: Backwards compatibility fix for default 'kms_data_key_reuse_period_seconds'.
- Add CHANGELOG entry.
- r/aws_sqs_queue: Acceptance tests passing.
- r/aws_sqs_queue_policy: Missing attribute equivalent to empty expected value.
- add on demand parmas
- allow updating `on_demand_target_capacity` and force new to other on demand arguments
- allow zero value on demand target capacity
- add default and validation for `on_demand_allocation_strategy`
- add docs
- add test for `on_demand_target_capacity`
- `on_demand_fulfilled_capacity` can change by other actions, adding computed add tests for on demand max price and allocation strategy
- add test for fulfilled capacity
- remove fulfilled capacity
- remove fulfilled capacity
- use enum slice
- rebase
- rebase
- validations
- add validations
- changelog
- error check
- fmt
- fmt
- r/aws_spot_fleet_request: Don't validate EC2 instance types.
- r/aws_spot_fleet_request: Instead of 'd.GetOkExists', always set in value. Tidy up 'resourceAwsSpotFleetRequestUpdate'.
- Update CHANGELOG.md for #19834
- Add CHANGELOG entry.
- r/lakeformation_perm: Fix update bug
- tests/r/lakeformation_perm: Test bug fix
- docs/r/lakeformation_perm: Update docs
- r/lakeformation_perm: Add changelog
- r/lakeformation_perms: Update changelog
- ds/lakeformation_perms: Fix list order diffs
- r/lakeformation_perms: Enable cross package use of filter
- tests/r/lakeformation_perms: Rework twc tests
- tests/lakeformation: Add disappears
- i/lakeformation: Use statuser
- docs/ds/lakeformation_perms: Update docs
- r/lakeformation_perms: Appease linter gods
- i/lakeformation: Simplify logic
- tests/r/lakeformation_permissions: Fix tests
- tests/lakeformation: Break into better groupings
- tests/lakeformation_perms: Appease linter gods
- r/aws_lakeformation_permissions: Fix 'testAccAWSLakeFormationPermissions_dataLocation'.
- i/lakeformation: Add filter unit tests
- r/aws_eks_node_group: Test that 0 is allowed for 'desired_size' and 'min_size'.
- i/lakeformation: Add explanation, extra safety check
- i/lakeformation: Add filter unit tests
- i/lakeformation: Fix comments
- i/lakeformation: Fix comments
- Update CHANGELOG.md for #19817
- r/aws_eks_node_group: Use supported Kubernetes versions in acceptance tests.
- updated route table conflict warning (#19852)
- provider: Add new data source
- ds/servicecat_portfolio: New data source
- tests/ds/servicecat_portfolio: New data source
- docs/ds/servicecat_portfolio: New data source
- ds/servicecat_portfolio: Add changelog
- ds/servicecat_portfolio: Make tags computed
- ds/servicecat_port: Fix changelog
- ds/servicecat_port: Minor fixes
- docs/ds/servicecat_port: Fix docs
- Update CHANGELOG.md for #19831
- Update CHANGELOG.md for #19500
- tests/r/waf_byte_match_set: Add sweeper concurrency
- tests/r/waf_geo_match_set: Add sweeper concurrency
- tests/r/waf_ipset: Add sweeper concurrency
- tests/r/waf_byte_match_set: Add read concurrency
- tests/r/waf_geo_match_set: Add read concurrency
- tests/r/waf_ipset: Add read concurrency
- tests/r/waf_rate_based_rule: Add sweeper concurrency
- tests/r/waf_regex_match_set: Add sweeper concurrency
- tests/r/waf_regex_pattern_set: Add sweeper concurrency
- tests/r/waf_rule_group: Add sweeper concurrency
- tests/r/waf_rule: Add sweeper concurrency
- tests/r/waf_size_constraint_set: Add sweeper concurrency
- tests/r/waf_sql_injection_match_set: Add sweeper concurrency
- tests/r/waf_web_acl: Add read/sweeper concurrency
- tests/r/waf_xss_match_set: Add sweeper concurrency
- tests/r/waf: Consistent var list
- tests/r/waf: Use consistent var name
- tests/r/waf: Remove unneeded len check
- Add CHANGELOG entry
- Update CHANGELOG.md for #19820
- v3.46.0





### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request




Relates OR Closes #0000

Output from acceptance testing:


```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```
